tests/php/ORM/DataObjectTest/DataListQueryCounter.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace SilverStripe\ORM\Tests\DataObjectTest;
+
+use SilverStripe\ORM\DataQuery;
+
+/**
+ * This is designed around the chunk method so we can count the number of queries run.
+ */
+class DataListQueryCounter extends DataQuery
```

Comment:
Does it make sense for this class to be under ORM? Should it not be under `Tests` or `Dev` since it doesn't really have any inherent production value?

Reply:
It's in `tests/`, which I think is fine. Maybe just implement `TestOnly` for good measure?

Reply:
Looks like adding `implements TestOnly` didn't eventuate before the merge.

Reply:
We discussed it offline and determined it wasn't of value. Did you have any thoughts on it?

Reply:
Not really, just looked unresolved without noting the outcome of the offline convo.